main/unwindi.c

```diff
@@ -203,21 +203,23 @@ CTAGS_INLINE MIOPos uugcGetFilePosition (void)
 		return getInputFilePosition ();
 }
 
-static ptrArray *uwiMarkerStack;
-static ptrArray *uwiCurrentMarker;
+static ptrArray *uwiBuffer;
+static unsigned int uwiMarkerStack[256];
```

Comment:
Is this (256) is safe enough?
If the answer is yes, we don't need
```
Assert (uwiCurrentMarker - uwiMarkerStack < 255);
```

Consider ctags runs as part of a network service. I'm afraid this 256 may help an attacher design an evil input.

Reply:
256 is hardcoded. At least, it should be defined as a macro.
So you can reuse the macro in other areas in this source file.

An alternative approach is that passing the limit to uwi layer from client side (typescript.c) via uwiActivate like
```
   uwiActivate(512);
```

Reply:
> Is this (256) is safe enough?

It depends on the parser implementation. Asserts here only aim to help the developer with debugging.

> 
> Consider ctags runs as part of a network service. I'm afraid this 256 may help an attacher design an evil input.

You're probably right but the attack would be more related to how parser works.

Reply:
> 256 is hardcoded. At least, it should be defined as a macro.
> So you can reuse the macro in other areas in this source file.

Ok, macro is a good idea!

> An alternative approach is that passing the limit to uwi layer from client side (typescript.c) via uwiActivate like
> 
> ```
>    uwiActivate(512);
> ```

And that's even better idea, thanks!